brainsets/utils/split.py

```
@@ -325,3 +364,198 @@ def generate_stratified_folds(
+    ...     assignments = generate_subject_kfold_assignment(subject_id)
+    ...     session = load_session(subject_id)
+    ...     for key, value in assignments.items():
+    ...         setattr(session, key, value)
+
+    See Also
+    --------
+    generate_trial_folds_by_task : For intra-session (trial-level) splitting.
+    """
+    subject_str = f"{subject_id}_{seed}"
+    subject_bytes = subject_str.encode("utf-8")
+    hash_obj = hashlib.md5(subject_bytes)
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)